wagtail/core/models/__init__.py

```diff
@@ -3680,12 +3470,13 @@ def save(self, user=None, specific_instance=None, **kwargs):
                 }
             )
 
-    def delete(self, user=None, specific_instance=None, **kwargs):
+    def delete(self, user=None, **kwargs):
```

Comment:
Not quite sure of the consequences of this change. It might be safer to keep functional changes out of this PR?

Reply:
I tried to keep the code changes to a minimum, but this one was needed so that BaseViewRestriction didn't have a dependency on PageLogEntry (which would prevent us from moving it out into its own module). I don't think there's a functional change here - the only reasonable circumstance where BaseViewRestriction.save and BaseViewRestriction.delete would be passed a `specific_instance` kwarg is in the `super` call within PageViewRestriction. If anything, I'd say that the presence of page-related code in BaseViewRestriction rather than PageViewRestriction was a (benign) bug.

Reply:
Ah, I see that now. Thanks for clarifying!